src/hooks/globalContexts/modDownloadUrl/useModDownloadUrl.ts

```diff
@@ -0,0 +1,109 @@
+import axios from "axios";
+import { useContext, useEffect, useState } from "react";
+import { getNewestFileIdFromGameBanana } from "../../gamebananaApi/getNewestFileIdFromGameBanana";
+import type { ModSearchDatabase_ModInfo_File } from "~/server/gamebananaMirror/yamlHandlers/modSearchDatabase/constAndTypes";
+import { GAMEBANANA_MOD_DOWNLOAD_BASE_URL, type NewestFileId } from "./constAndTypes";
+import { modDownloadUrlContext } from "./modDownloadUrlContext";
+import { getNewestFileIdFromModSearchDatabaseModFiles } from "~/server/gamebananaMirror/yamlHandlers/modSearchDatabase/getNewestFileIdFromModSearchDatabaseModFiles";
+
+
+
+
+const getModDownloadUrl = (
+    gamebananaModId: number,
+    newestFileId: NewestFileId,
+) => {
+    const downloadUrl = newestFileId === "" ? "" : `${GAMEBANANA_MOD_DOWNLOAD_BASE_URL}${newestFileId},Mod,${gamebananaModId}`;
+
+    return downloadUrl;
+};
+
+
+
+
+export type UseModDownloadUrlProps = {
+    gamebananaModId: number,
+    filesFromModSearchDatabase: ModSearchDatabase_ModInfo_File[] | undefined,
+};
+
+
+export const useModDownloadUrl = (
+    {
+        gamebananaModId,
+        filesFromModSearchDatabase,
+    }: UseModDownloadUrlProps,
+): string => {
+    const contextOrUndefined = useContext(modDownloadUrlContext);
+
+    const cachedDownloadUrl = contextOrUndefined?.state[gamebananaModId];
+
+    const [downloadUrl, setDownloadUrl] = useState<string>(cachedDownloadUrl ?? "");
```

Comment:
Do we need this state? We update the context whenever we update the downloadUrl, so couldn't we just use the cachedDownloadUrl?

Reply:
Yeah I think we can. I'll take a stab at it.